docs/guides/datamigrations/postgres.rst

```diff
@@ -0,0 +1,574 @@
+
+.. lint-on
+
+The ``badges`` table actually uses the ``name`` of the badge as a primary key
+as opposed to a separate ``id``. So in order to reflect that properly in
+EdgeDB we would have to add :eql:constraint:`exclusive` constraint to this
+property. Meanwhile ``not null`` makes the property ``required``, so we end up
+with a type like this:
+
+.. code-block:: sql
+
+    type Badge {
+        required name: str {
+            constraint exclusive;
+        }
+        required description: str;
+    }
+
+The ``statuses`` table actually has a dedicated ``id`` column in addition to
+``title``. However, ``id`` in EdgeDB is a :eql:type:`uuid`, whereas in our
+original dataset it is an ``integer``. Luckily, the ``title`` is ``UNIQUE``
+and we might be able to use that to correctly map the data during our
+migration and leave the old ``id`` entirely behind (assuming that no code
+actually depends on it). This leaves us with the following type:
+
+.. code-block:: sql
+
+    type Status {
+        required title: str {
+            constraint exclusive;
+        }
+    }
+
+Next we can look at the ``users`` table:
+
+.. lint-off
+
+.. code-block::
+
+    social=> \d users
+                                     Table "public.users"
+         Column      |  Type   | Collation | Nullable |              Default
+    -----------------+---------+-----------+----------+-----------------------------------
+     id              | integer |           | not null | nextval('users_id_seq'::regclass)
+     name            | text    |           | not null |
+     email           | text    |           | not null |
+     password        | text    |           | not null |
+     client_settings | jsonb   |           |          |
+     badge_name      | text    |           |          |
+     status_id       | integer |           |          |
+    Indexes:
+        "users_pkey" PRIMARY KEY, btree (id)
+        "users_email_key" UNIQUE CONSTRAINT, btree (email)
+        "users_name_key" UNIQUE CONSTRAINT, btree (name)
+    Foreign-key constraints:
+        "users_badge_name_fkey" FOREIGN KEY (badge_name) REFERENCES badges(name)
+        "users_status_id_fkey" FOREIGN KEY (status_id) REFERENCES statuses(id)
+    Referenced by:
+        TABLE "bookmarks" CONSTRAINT "bookmarks_user_id_fkey" FOREIGN KEY (user_id) REFERENCES users(id)
+        TABLE "posts" CONSTRAINT "posts_user_id_fkey" FOREIGN KEY (user_id) REFERENCES users(id)
+
+.. lint-on
+
+The ``users`` table also has an ``id`` column, which is not a
+:eql:type:`uuid`. Instead of omitting this information we'll record it as
+``app_id`` to facilitate the transition. Potentially we still want to
+eventually drop it in favor of the built-in ``id`` from EdgeDB, but we need it
+for now. The column ``client_settings`` would become a :eql:type:`json`
+property. The columns ``badge_name`` and ``status_id`` reference ``badges``
+and ``statuses``, respectively and will become *links* instead of
+*properties*:
```

Comment:
```suggestion
The ``users`` table, like ``statuses``, has an ``id`` column, which is not a
:eql:type:`uuid`. Instead of omitting the ``id`` data, we'll record it as
``app_id`` in EdgeDB to facilitate the transition. We may still want to
eventually drop it in favor of the built-in ``id`` from EdgeDB, but we need it
for now. The column ``client_settings`` would become a :eql:type:`json`
property. The columns ``badge_name`` and ``status_id`` reference ``badges`` and
``statuses`` respectively and will become *links* in EdgeDB instead of
*properties*, even though a property would more closely mirror how they are
stored in Postgres:
```